fastavro/_write.pyx

```diff
@@ -112,7 +114,7 @@ cdef inline write_double(bytearray fo, double datum):
     fo += ch_temp[:8]
 
 
-cdef inline write_bytes(bytearray fo, bytes datum):
+cdef inline write_bytes(bytearray fo, const unsigned char[:] datum):
```

Comment:
I'm not too familiar with all the different types and how cython uses them. Is there a noticeable impact from this change?

Reply:
This is what makes it possible to use both bytearrays and bytes.
It is also recommended [here](https://cython.readthedocs.io/en/latest/src/tutorial/strings.html#accepting-strings-from-python-code)

> In the case that the API only deals with byte strings, i.e. binary data or encoded text, it is best not to type the input argument as something like bytes, because that would restrict the allowed input to exactly that type and exclude both subtypes and other kinds of byte containers, e.g. bytearray objects or memory views.

You do lose out on performance, which is why I removed the function call from `write_utf8`.